### ISSUE TITLE
flutter doctor reports fine in terminal but fails in VSCode (Run from command palette) 

### ISSUE THREAD

**Issue by Pager07** [flutter/flutter#96765]
## Steps to Reproduce
1. Running `flutter doctor` from terminal (works fine)
![image](https://user-images.githubusercontent.com/21243997/149909286-b7ad41f7-762f-4b89-991a-00d60611169d.png)

2. Running `flutter doctor` from vscode terminal (works fine)
![image](https://user-images.githubusercontent.com/21243997/149909614-8614d46a-a85c-4b2a-8015-707d1fa587d0.png)
3.  Running flutter doctor from vscode command palette (error)
![image](https://user-images.githubusercontent.com/21243997/149909798-9b90505c-5de8-4b04-9d5a-d12dc6c8b337.png)

## Logs
I have followed this [issue](https://github.com/flutter/flutter/issues/19703) to log dart.extensionLog. It seems that the paths to flutter are correct given that it finds Flutter and SDK. Hence, not sure what lead to the error.
```
Dart Code extension: 3.33.20220111
Flutter extension: 3.32.0 (not activated)

App: Visual Studio Code
Version: 1.63.2
Platform: mac

HTTP_PROXY: undefined
NO_PROXY: undefined

Logging Categories:
    General

Tue Jan 18 2022 [09:17:40 GMT+0000 (Greenwich Mean Time)] Log file started
[9:17:40 AM] [General] [Info] Searching for SDKs...
[9:17:40 AM] [General] [Info] Environment PATH:
[9:17:40 AM] [General] [Info]     /opt/local/bin
[9:17:40 AM] [General] [Info]     /opt/local/sbin
[9:17:40 AM] [General] [Info]     /Users/sandeep/opt/anaconda3/bin
[9:17:40 AM] [General] [Info]     /Users/sandeep/opt/anaconda3/condabin
[9:17:40 AM] [General] [Info]     /Users/sandeep/.local/bin
[9:17:40 AM] [General] [Info]     /Library/Frameworks/Python.framework/Versions/3.7/bin
[9:17:40 AM] [General] [Info]     /Users/sandeep/anaconda3/bin
[9:17:40 AM] [General] [Info]     /Library/Frameworks/Python.framework/Versions/3.5/bin
[9:17:40 AM] [General] [Info]     /Library/Frameworks/Python.framework/Versions/3.6/bin
[9:17:40 AM] [General] [Info]     /Library/Frameworks/Python.framework/Versions/3.6/bin
[9:17:40 AM] [General] [Info]     /usr/local/bin
[9:17:40 AM] [General] [Info]     /usr/bin
[9:17:40 AM] [General] [Info]     /bin
[9:17:40 AM] [General] [Info]     /usr/sbin
[9:17:40 AM] [General] [Info]     /sbin
[9:17:40 AM] [General] [Info]     /Library/TeX/texbin
[9:17:40 AM] [General] [Info]     /opt/X11/bin
[9:17:40 AM] [General] [Info]     /Library/Apple/usr/bin
[9:17:40 AM] [General] [Info]     /Library/Frameworks/Mono.framework/Versions/Current/Commands
[9:17:40 AM] [General] [Info]     /Applications/Wireshark.app/Contents/MacOS
[9:17:40 AM] [General] [Info]     /Users/sandeep/Documents/devs/flutter/bin
[9:17:40 AM] [General] [Info] Found Flutter project at /Users/sandeep/Desktop/flutterapp/simpleinterestcalc/calc:
			Mobile? false
			Web? false
			Create Trigger? true
			Flutter Repo? false
[9:17:40 AM] [General] [Info] Searching for flutter
[9:17:40 AM] [General] [Info]     Looking for flutter in:
[9:17:40 AM] [General] [Info]         /Users/sandeep/Documents/flutter
[9:17:40 AM] [General] [Info]         /Users/sandeep/Documents/flutter/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/Desktop/flutterapp/simpleinterestcalc/calc
[9:17:40 AM] [General] [Info]         /Users/sandeep/Desktop/flutterapp/simpleinterestcalc/calc/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/Desktop/flutterapp/simpleinterestcalc/calc/.flutter
[9:17:40 AM] [General] [Info]         /Users/sandeep/Desktop/flutterapp/simpleinterestcalc/calc/.flutter/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/Desktop/flutterapp/simpleinterestcalc/calc/vendor/flutter
[9:17:40 AM] [General] [Info]         /Users/sandeep/Desktop/flutterapp/simpleinterestcalc/calc/vendor/flutter/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/flutter-sdk
[9:17:40 AM] [General] [Info]         /Users/sandeep/flutter-sdk/bin
[9:17:40 AM] [General] [Info]         /google/flutter
[9:17:40 AM] [General] [Info]         /google/flutter/bin
[9:17:40 AM] [General] [Info]         /opt/local/bin
[9:17:40 AM] [General] [Info]         /opt/local/sbin
[9:17:40 AM] [General] [Info]         /Users/sandeep/opt/anaconda3/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/opt/anaconda3/condabin
[9:17:40 AM] [General] [Info]         /Users/sandeep/opt/anaconda3/condabin/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/.local/bin
[9:17:40 AM] [General] [Info]         /Library/Frameworks/Python.framework/Versions/3.7/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/anaconda3/bin
[9:17:40 AM] [General] [Info]         /Library/Frameworks/Python.framework/Versions/3.5/bin
[9:17:40 AM] [General] [Info]         /Library/Frameworks/Python.framework/Versions/3.6/bin
[9:17:40 AM] [General] [Info]         /Library/Frameworks/Python.framework/Versions/3.6/bin
[9:17:40 AM] [General] [Info]         /usr/local/bin
[9:17:40 AM] [General] [Info]         /usr/bin
[9:17:40 AM] [General] [Info]         /bin
[9:17:40 AM] [General] [Info]         /usr/sbin
[9:17:40 AM] [General] [Info]         /sbin
[9:17:40 AM] [General] [Info]         /Library/TeX/texbin
[9:17:40 AM] [General] [Info]         /Library/TeX/texbin/bin
[9:17:40 AM] [General] [Info]         /opt/X11/bin
[9:17:40 AM] [General] [Info]         /Library/Apple/usr/bin
[9:17:40 AM] [General] [Info]         /Library/Frameworks/Mono.framework/Versions/Current/Commands
[9:17:40 AM] [General] [Info]         /Library/Frameworks/Mono.framework/Versions/Current/Commands/bin
[9:17:40 AM] [General] [Info]         /Applications/Wireshark.app/Contents/MacOS
[9:17:40 AM] [General] [Info]         /Applications/Wireshark.app/Contents/MacOS/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/Documents/devs/flutter/bin
[9:17:40 AM] [General] [Info]     Found at:
[9:17:40 AM] [General] [Info]         /Users/sandeep/Documents/flutter/bin
[9:17:40 AM] [General] [Info]         /Users/sandeep/Documents/devs/flutter/bin
[9:17:40 AM] [General] [Info]     Candidate paths to be post-filtered:
[9:17:40 AM] [General] [Info]         /Users/sandeep/Documents/flutter
[9:17:40 AM] [General] [Info]         /Users/sandeep/Documents/devs/flutter
[9:17:40 AM] [General] [Info]     Found at /Users/sandeep/Documents/flutter
[9:17:40 AM] [General] [Info]     Returning SDK path /Users/sandeep/Documents/flutter for flutter
[9:17:40 AM] [General] [Info] Flutter is not initialized, running 'flutter doctor' to force...
[9:17:40 AM] [General] [Info] (PROC 22444) Logging data for process...
[9:17:40 AM] [General] [Info] (PROC 22444) fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
[9:17:40 AM] [General] [Info] (PROC 22444) Building flutter tool...
[9:17:41 AM] [General] [Info] (PROC 22444) Error on line 1, column 1 of pubspec.yaml: Unexpected character.
  ╷
1 │ ������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������…
[9:17:42 AM] [General] [Info] (PROC 22444) Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
[9:17:47 AM] [General] [Info] (PROC 22444) Error on line 1, column 1 of pubspec.yaml: Unexpected character.
[9:17:47 AM] [General] [Info] (PROC 22444)
[9:17:47 AM] [General] [Info] (PROC 22444)   ╷
[9:17:47 AM] [General] [Info] (PROC 22444) 1 │ ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������…
[9:17:47 AM] [General] [Info] (PROC 22444) 
  │ ^
  ╵
[9:17:47 AM] [General] [Info] (PROC 22444) Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (8 tries left)
[9:17:52 AM] [General] [Info] (PROC 22444) Error on line 1, column 1 of pubspec.yaml: Unexpected character.
[9:17:52 AM] [General] [Info] (PROC 22444)

```
<details>
<summary>Logs</summary>
flutter command works fine in terminal and vscode terminal instance. But the fultter commands leads to error with vscode command palette.

```
```

```
```

</details>

**Comment by maheshmnj** [comment]
Hi @Pager07, Thanks for filing the issue. could you please share the output of  `which flutter` running from  vscode and the mac terminal, Please ensure that they both have the same path

> Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds..

This looks like the tool is unable to download from pub, Please make sure you have a working internet connection and if it is, then please try deleting the cache folder located in `flutter/bin/cache`

and then from the terminal run `flutter` and let it download the dart SDK.

**Comment by Pager07** [comment]
I tried following the guide from @maheshmnj , where I deleted the `fluttter/bin/cache` and ran `fultter` it started downloading the dart sdk but it again failed when at `Building flutter tool....` stage. 

But the issued is now fixed, and here is how I fixed it.
- I removed the following 3 files; `flutter`,`.flutter` and `.dart` (run `ls -a` in your $HOME or the dir where `flutter` is placed to find these files)
- I initially downloaded flutter in `~/Documents`. Redownloaded the flutter from git in `$HOME` dir following: [HERE](https://docs.flutter.dev/get-started/install/macos)(Note, last time around, I skipped the `flutter precache` step, but I didn't this time.
- Restarted terminal, vscode. 

Now working like a charm. 

Summary:
- Problem is fixed but doesn't really know what caused the problem.
